### PR TITLE
Support nest incomplete type

### LINKED ISSUE #105: Invalid thrift_spec if type is used before define
Codes to reproduce this:

test.thrift

```thrift
struct Container {
    1: A field1;
    2: list<A> field2;
    3: list<map<string, A>> field3;
}

struct A {
    1: string value
}
```

test.py

```python
import thriftpy2
from pprint import pprint

test_thrift = thriftpy2.load('./test.thrift')

pprint(test_thrift.Container.thrift_spec)
```

output:

```
{1: (12, 'field1', <class 'test.A'>, False),
 2: (15, 'field2', (12, <class 'test.A'>), False),
 3: (15, 'field3', (13, (11, -3)), False)}
```

Notice the `field3`, it's spec is expected to be `(15, 'field3', (13, (11, <class 'test.A'>)), False)` like `field2`.

### PR DESCRIPTION
**Pull request by aisk**
Try to fix #105 .

### CODE REVIEW

File: thriftpy2/parser/__init__.py [hunk 1]
```diff
@@ -111,6 +111,18 @@ def fill_incomplete_ttype(tmodule, definition):
                         )
                     ),
                     value[3])
+            elif isinstance(value[2], tuple):
```
Comment (ethe) [inline thriftpy2/parser/__init__.py]: It is better to leave some comments here.

**Comment by ethe** [comment]
`fill_incomplete_ttype ` has become too hard to understand, it is better to leave some comments there, maybe I will refactor it in the future.

**Comment by aisk** [comment]
@ethe updated

**Comment by ethe** [comment]
LGTM

**Pull request merged**